UnitTestFrameworkPkg/GoogleTest.md

```diff
@@ -742,9 +786,17 @@ With those limitations in mind, that completes the mock function
 declarations, and now the mock function definitions for those declarations
 can be created.
 
-### FunctionMockLib Mocks - Definition
 
-The definition of mock functions using the FunctionMockLib macros are done
+### FunctionMockLib Mocks - Definition
+The definition of mock functions using FunctionMockLib is done
+differently for library and non-library function mocks. Library mock
+functions are built using their own separate INF file and other mock
+functions (of a protocol, for example) are built in the source file of each GoogleTest

```

Comment:
Change to `(of a protocol or any other service table with function pointers)`.  That should cover all the cases.